ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -28,16 +31,144 @@ const yeheheOutputStrings = {
+  FL_BR = (TtokSafeSpots.Front & TtokSafeSpots.Left) | (TtokSafeSpots.Back & TtokSafeSpots.Right),
+  InRight = TtokSafeSpots.In & TtokSafeSpots.Right,
+  InLeft = TtokSafeSpots.In & TtokSafeSpots.Left,
+}
+
+const ttokrroneTempestSandspoutOutputStrings = {
+  unknown: Outputs.unknown,
+  in: Outputs.getUnder,
+  out: Outputs.outOfMelee,
+  outOfHitbox: Outputs.outOfHitbox,
+  right: Outputs.right,
+  left: Outputs.left,
+  back: Outputs.back,
+  front: Outputs.front,
+  rear: {
+    en: 'rear',
+  },
+  rightFlank: {
+    en: 'right flank',
+  },
+  leftFlank: {
+    en: 'left flank',
+  },
+  triple: {
+    en: '${inOut} + ${dir2} + ${dir3}',
+  },
+  double: {
+    en: '${inOut} + ${dir2}',
+  },
+  awayFrom: {
+    en: '${out} + avoid ${dir}',
+  },
+} as const;
+
+type Point = { x: number; y: number };
+type TtokrroneDodge = 'in' | 'out' | 'right' | 'left' | 'front' | 'back' | 'unknown';
+const ttokrroneDesertTempest: { [id: string]: TtokSafeSpots } = {
+  '91D3': TtokSafeSpots.Out,
+  '91D4': TtokSafeSpots.In,
+  '91D5': TtokSafeSpots.InRight,
+  '91D6': TtokSafeSpots.InLeft,
+};
+const ttokrroneDesertTempestIds = Object.keys(ttokrroneDesertTempest);
+
+const ttokrroneDustdevilOutputStrings = {
+  outOfHitbox: {
+    en: 'Out of hitbox + stay out',
+  },
+  rotateFront: {
+    en: 'Rotating frontal cleave', // ${dir}'
+  },
+  rotateRear: {
+    en: 'Rotating rear cleave', // ${dir}'
+  },
+} as const;
+
+const identifyOrbSafeSpots = (pattern: Point[]) => {
+  // Each time Ttokrrone summons sand orbs either 7 or 8 are spawned in 4 patterns.
+  // (these are relative to where the boss faces, which is south)
+  // 7 covering all rel. north
+  // 7 covering all rel. south
+  // 8 covering the rel. NE and SW quadrants with some overlap in the middle
+  // 8 covering the rel. NW and SE quadrants with some overlap in the middle
+  // since each pattern is composed of a quadrant of AOEs we can check for any orbs in each quadrant
+  // (centered around the arena center) and then check the final result to determine the pattern.
+  let topRight = false;
+  let topLeft = false;
+  let bottomRight = false;
+  let bottomLeft = false;
+  for (const point of pattern) {
+    const xGt = point.x > 53.0;
+    const yGt = point.y > -825.0;
+    topLeft = !topLeft ? !xGt && !yGt : topLeft;
+    topRight = !topRight ? xGt && !yGt : topRight;
+    bottomRight = !bottomRight ? !xGt && yGt : bottomRight;
+    bottomLeft = !bottomLeft ? xGt && yGt : bottomLeft;
+  }
+  // Returns the SAFE SPOTS so opposite of what is detected
+  if (topLeft && topRight) { // AOEs all bottom
+    return TtokSafeSpots.Front;
+  }
+  if (bottomRight && bottomLeft) { // AOEs all top
+    return TtokSafeSpots.Back;
+  }
+  if (topLeft && bottomLeft) { // AOEs are front left, back right
+    return TtokSafeSpots.FR_BL;
+  }
+  if (topRight && bottomRight) { // AOEs are front right, back left
+    return TtokSafeSpots.FL_BR;
+  }
```

Comment:
Same here - looks like `bottomLeft` and `bottomRight` are swapped based on these comments as well.